modules/extension/wmts/src/main/java/org/geotools/ows/wmts/map/WMTSCoverageReader.java

```diff
@@ -259,33 +227,27 @@ GridCoverage2D getMap(
         GridCoverage2D result;
         if (isNativelySupported(requestedEnvelope.getCoordinateReferenceSystem())) {
             // we can simply perform the tile request and build the final image.
-            ReferencedEnvelope gridEnvelope =
+            TileRequest request =
                     initTileRequest(
                             requestedEnvelope,
                             readParameters.getWidth(),
                             readParameters.getHeight(),
                             time);
-            result = getMap(requestedEnvelope, gridEnvelope);
+
+            result = createTileMap(request, requestedEnvelope, time);
         } else {
             result = getMapReproject(requestedEnvelope, time, readParameters);
         }
         return result;
     }
 
-    private GridCoverage2D getMap(ReferencedEnvelope envelope, ReferencedEnvelope gridEnvelope)
+    GridCoverage2D createTileMap(TileRequest request, ReferencedEnvelope tileEnvelope, String time)
             throws IOException {
         try {
-            if (LOGGER.isLoggable(Level.FINE)) {
-                LOGGER.log(
-                        Level.FINE,
-                        "Issuing request: " + getTileRequest().getFinalURL(),
-                        new RuntimeException("TRACE!"));
```

Comment:
This exception bit was truly bad, but the logging of the request was useful. Is it happening anywhere else?

Reply:
I was just tired of that trace. Never considered the log message itself, but I've reintroduced it at the very end of the handling within WMTSCoverageReader.